### ISSUE TITLE
[BUG] S3 Secret Access Key Is Exposed in Plain Text on Settings

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#4360]
**Describe the bug**
Secret Access Key is exposed in plain text for backup settings w/ S3

**To Reproduce**
1. Create S3 Backup Endpoint
1. Click "show backup-target"
1. Secret Access Key is visible in plain text
 
**Expected behavior**
Secret Access Key to be obfuscated from user, not in plain text

**Environment**
 - Harvester ISO version: v1.2-head /  Version: v1.2-a40c4f2f-head 
 - Underlying Infrastructure: qemu/kvm

**Additional context**
![Screenshot from 2023-08-01 14-49-28](https://github.com/harvester/harvester/assets/5370752/7d34b6bc-fc8e-45a2-bcb2-edb917421fa9)

**Comment by irishgordo** [comment]
MinIO seems to by default obfuscate it:
![Screenshot from 2023-08-01 14-55-48](https://github.com/harvester/harvester/assets/5370752/dd4a0f2a-82d4-4c6e-a75d-4509bc327441)
![Screenshot from 2023-08-01 14-55-17](https://github.com/harvester/harvester/assets/5370752/e86c2577-7192-4e01-9a58-fab2e5a78c6f)

**Comment by irishgordo** [comment]
We obfuscate the Secret Access Key on the creation, but not on the viewing:

![Screenshot from 2023-08-01 14-57-04](https://github.com/harvester/harvester/assets/5370752/787d8388-c9a1-4427-bca5-0c07d5c58554)

**Comment by irishgordo** [comment]
Error message also does not obfuscate:
![Screenshot from 2023-08-01 15-10-41](https://github.com/harvester/harvester/assets/5370752/0cc73e0a-a7be-4430-85a3-22e462607ceb)


*please ignore the https error, as this is expected, doing some negotiation of endpoint currently

**Comment by irishgordo** [comment]
Harvester Version, v1.1.2 displays this, in contrast:
![Screenshot from 2023-08-02 14-31-16](https://github.com/harvester/harvester/assets/5370752/94f6e5e8-6063-4694-a798-7064d1a32bb4)

**Comment by irishgordo** [comment]
@futuretea 
This seems to be most triggered in a few different scenarios:
(tested v1.2.0-rc5)

Scenario A:
- for the "first-time" / "original" time in adding a Minio Connection with Custom `additional-ca` cert, ensure that when pasting in the cert, add extra new lines
- then fill out the rest for the backup-target, the region, access key, secret access key
- it will throw an error like:
```
admission webhook "[validator.longhorn.io](http://validator.longhorn.io/)" denied the request: failed to set the setting backup-target-credential-secret with invalid value harvester-backup-target-secret: there is space or new line in AWS_CERT
```
- the information will then be leaked to the front-end

The fix:
- remove the new lines in the cert, resave
- modify the backup-target, resave

Note, Scenario B seems harder to trigger, the order of operations may be changed to experiment surrounding triggering it as it doesn't always seem to work.
Scenario B:
- modify the cert to be empty, save
- test connection on backup-target, it should fail, w/ X509 errors
- build a new access-key & secret-access key, resave the backup-target with those
- paste back in the cert, into the `additonal-ca` but add extra new-lines, resave
- the information will then be leaked to the front-end

**Comment by irishgordo** [comment]
**validation issue**

When testing with  Version: v1.2-95dec1da-head  - it seems that I'm still able to leak the credentials out in plain text to the front-end, though refreshing the page does seem to fix that - perhaps websocket related?

@futuretea - with some variations on Scenario B, I'm still able to leak the credentials when the x509 issues creep up, but the x509, error message doesn't anymore show the credentials which is good :smile: 

![Screenshot from 2023-08-28 17-07-52](https://github.com/harvester/harvester/assets/5370752/da075f60-f075-4b7d-b1be-c823355190e8)
![Screenshot from 2023-08-28 17-05-17](https://github.com/harvester/harvester/assets/5370752/44cb4e39-90fa-4171-944a-86784e2673da)
[supportbundle_6335f71e-a856-4a23-9088-87dff11e1011_2023-08-29T00-10-10Z.zip](https://github.com/harvester/harvester/files/12458977/supportbundle_6335f71e-a856-4a23-9088-87dff11e1011_2023-08-29T00-10-10Z.zip)

**Comment by futuretea** [comment]
Hi @irishgordo, Did you try refreshing the page on Scenario B? The harvester pod restarts because the `additonal-ca` has just been updated, so the`backup-target` processing takes a while.

**Comment by WuJun2016** [comment]
> Note, Scenario B seems harder to trigger, the order of operations may be changed to experiment surrounding triggering it as it doesn't always seem to work.
Scenario B:
modify the cert to be empty, save
test connection on backup-target, it should fail, w/ X509 errors
build a new access-key & secret-access key, resave the backup-target with those
paste back in the cert, into the additonal-ca but add extra new-lines, resave
the information will then be leaked to the front-end

Hi @irishgordo  This scenario shows that it may be caused by the UI not using the latest websocket's data.
There are two scenarios:
1. the websocket disconnects and reconnects due to a change in the additional-ca, then if you configure the backup-target before it connects successfully, it will display the AKSK value on the UI.  If you refresh the page, the value will disappear.  (This should work as expected.)

2. websocket connection succeeded and pushed data to the frontend. But the UI is not rendering correctly. (UI [bug](https://github.com/harvester/harvester/issues/4463), planned to be fixed in v1.2.1)   cc @guangbochen   @futuretea

**Comment by irishgordo** [comment]
@futuretea  you're correct :smile:  - I've noticed that yes, absolutely refreshing the page on Scenario B fixes it all.

@WuJun2016 thanks for that info as well :smile:  - since 4463 has been opened to track that issue I think that's sufficient :+1: 

Thank you both - I'll go ahead and close this out - thanks again for the fix @futuretea :smile:
